bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/QuantityType.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.library.types;
+
+import java.math.BigDecimal;
+
+import javax.measure.Dimension;
+import javax.measure.Quantity;
+import javax.measure.UnconvertibleException;
+import javax.measure.Unit;
+import javax.measure.UnitConverter;
+
+import org.eclipse.smarthome.core.library.internal.StateConverterUtil;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.Convertible;
+import org.eclipse.smarthome.core.types.PrimitiveType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+
+import tec.uom.se.AbstractUnit;
+import tec.uom.se.quantity.Quantities;
+
+/**
+ * The measure type extends DecimalType to handle physical unit measurement
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+public class QuantityType extends Number
+        implements PrimitiveType, State, Command, Comparable<QuantityType>, Convertible {
+
+    private static final long serialVersionUID = 8828949721938234629L;
+    private static final BigDecimal HUNDRED = BigDecimal.valueOf(100);
+
+    // Regular expression to split unit from value
+    private static final String UNIT_PATTERN = "(?<=\\d)\\s*(?=[a-zA-Z°µ])";
+
+    protected Quantity<?> quantity;
+
+    public QuantityType(String value) {
+        if (value != null) {
+            String[] constituents = value.split(UNIT_PATTERN);
+
+            // getQuantity needs a space between numeric value and unit
+            String formatted = String.join(" ", constituents);
+            quantity = Quantities.getQuantity(formatted);
+        } else {
+            throw new IllegalArgumentException("Constructor argument must not be null");
```

Comment:
Can you turn this around and return early with the IAE? Then the happy path needs no condition.